pkg/auth/iguazio/v4/auth.go

```diff
@@ -17,103 +17,73 @@ limitations under the License.
 package v4
 
 import (
-	"context"
 	"encoding/json"
 	"fmt"
-	"io"
 	"net/http"
 
 	authpkg "github.com/nuclio/nuclio/pkg/auth"
 	"github.com/nuclio/nuclio/pkg/auth/iguazio"
+	"github.com/nuclio/nuclio/pkg/common"
 	"github.com/nuclio/nuclio/pkg/common/headers"
 
 	"github.com/nuclio/errors"
 	"github.com/nuclio/logger"
 	"github.com/nuclio/nuclio-sdk-go"
 )
 
-const OAuth2ProxyCookie = "_oauth2_proxy"
 const groupType = "type.googleapis.com/group.Group"
 
 type Auth struct {
 	*iguazio.AbstractAuth
 }
 
-func NewAuth(logger logger.Logger, config *authpkg.Config) (authpkg.Auth, error) {
-	return &Auth{
-		AbstractAuth: iguazio.NewAbstractAuth(logger, config),
-	}, nil
+func NewAuth(logger logger.Logger, config *authpkg.Config) authpkg.Auth {
+	authenticator := &Auth{}
+	authenticator.AbstractAuth = iguazio.NewAbstractAuth(logger, config, authenticator)
+	return authenticator
 }
 
-func (a *Auth) Authenticate(request *http.Request, options *authpkg.Options) (authpkg.Session, error) {
+func (a *Auth) VerifySessionType(session interface{}) (authpkg.Session, bool) {
+	if typedSession, ok := session.(*Session); ok {
+		return typedSession, true
+	}
+	return nil, false
+}
```

Comment:
This is the same as v1

Reply:
@TomerShor we need them in both as each transforms to its own session (v1/v4)